### ISSUE TITLE
Error while running watch-storefront script

### ISSUE THREAD

**Issue by hagmann** [dockware/dockware#11]
While running watch-storefront, the following error occurred:
![image](https://user-images.githubusercontent.com/11349379/93527176-c366cb00-f938-11ea-9c2f-a9607a46c647.png)

The compilation was still successful, but the JS files are not being loaded using port 9999:
![image](https://user-images.githubusercontent.com/11349379/93527128-b2b65500-f938-11ea-93ce-55c212baf74b.png)

This may lead to this error:
![image](https://user-images.githubusercontent.com/11349379/93527020-93b7c300-f938-11ea-9029-b2d9ded4198e.png)

The sources are available as expected on localhost:9999.

System details:
Windows 10 Pro
Docker Toolbox

**Comment by martinweinmayr** [comment]
@hagmann which dockware image are you running?

**Comment by hagmann** [comment]
@martinweinmayr 

Dockerfile:

```
version: "3"

services:

  shopware:
    image: dockware/dev:latest
    container_name: badminton-onlineshop
    ports:
      - "80:80"
      - "3306:3306"
      - "222:22"
      - "8888:8888"
      - "9999:9999"
    networks:
      - web
    environment:
      - SW_CURRENCY=CHF
      - XDEBUG_ENABLED=1
      - XDEBUG_REMOTE_HOST=192.168.99.101

networks:
  web:
    external: false
```

I started the watcher using 
```
cd /var/www
make watch-storefront
```

**Comment by martinweinmayr** [comment]
@hagmann which dockware version?

dev/contribute/essential?

shopware version? 6.3.0?

**Comment by hagmann** [comment]
@martinweinmayr 

dockware/dev:latest
Shopware 6.3.1.0 (Updated in the Administration)

**Comment by martinweinmayr** [comment]
just to be sure, in the .env you have SHOPWARE_HTTP_CACHE_ENABLED=0
?

**Comment by hagmann** [comment]
```
APP_ENV=dev
APP_SECRET=1
INSTANCE_ID=1
DATABASE_URL=mysql://root:root@localhost:3306/shopware
APP_URL=http://localhost
MAILER_URL=smtp://localhost:1025
COMPOSER_HOME=/var/www/html/var/cache/composer
SHOPWARE_ES_ENABLED=0

```

**Comment by dimitripreiss** [comment]
Hi,

I have the same problem after I updated Shopware to 6.3.1.0 in the administration ([#10](https://github.com/dockware/dockware/issues/10#issuecomment-692181943)). 

@martinweinmayr I checked your suggestion and add SHOPWARE_HTTP_CACHE_ENABLED = 0 but but unfortunately the error still exists.

**Comment by martinweinmayr** [comment]
we will pack sw 6.3.1 on monday, i think then we will also see what the problem ist ...
Normaly a admin update should not be that problem as it is based on production theme.

I'll give you feedback latest on monday, maybe already today

**Comment by martinweinmayr** [comment]
ps. on my site it works well... Strange

**Comment by martinweinmayr** [comment]
in the and of the process you see something like that?

![image](https://user-images.githubusercontent.com/13133302/93585618-5d655c80-f9a7-11ea-9f12-ab86fb5ba686.png)

and if you change a scss file it reloads automaticly? right?

**Comment by martinweinmayr** [comment]
ahhhh i see on the 1st image they try to open 9998.
This should not happen if you really use our make command. @hagmann 

If you do it your way you have to map the port 9998:9998 in your ports section of the docker-compose file.

**Comment by hagmann** [comment]
@martinweinmayr I actually followed the docs so I am not sure what it tries to use port 9998 instead of 9999. Both ports are now mapped in docker-compose.xml and forwared to the docker machine.

The compiler itself works flawlessy.

**Comment by martinweinmayr** [comment]
compiler and watcher works now? "compiled successfull" is also the result of watchers ...

**Comment by hagmann** [comment]
@martinweinmayr As mentioned in the first post, the compiler always worked. The issue are the resources that are not being loaded through port 9999.

**Comment by martinweinmayr** [comment]
if the watcher would work you should see the message "compiled successfull" and not what you have postet in your first image

**Comment by hagmann** [comment]
Here is a larger image of the steps and where it fails
![image](https://user-images.githubusercontent.com/11349379/93593899-2f871480-f9b5-11ea-97d0-a3af1fba5649.png)

**Comment by martinweinmayr** [comment]
mine lookes like this

![image](https://user-images.githubusercontent.com/13133302/93596589-e1c0db00-f9b9-11ea-96b4-bfd4c286ebc7.png)

**Comment by martinweinmayr** [comment]
especially this line seems me missing on your site:
cd /var/www/html && PROJECT_ROOT=/var/www/html APP_URL=http://localhost ESLINT_DISABLE=true npm --prefix /var/www/html/vendor/shopware/storefront/Resources/app/storefront run hot

**Comment by martinweinmayr** [comment]
can you check your makefile pls?

should look like this:
.PHONY: help
.DEFAULT_GOAL := help

help:
	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'

# ---------------------------------------------------------------------------------------------
watch-storefront: ## starts watcher for admin at http://localhost
	@echo "RequestHeader add hot-reload-mode "1"" > /var/www/html/.htaccess.watch
	cd /var/www/html && ./bin/build-storefront.sh
	cd /var/www/html && php bin/console theme:dump
	cd /var/www/html && PROJECT_ROOT=/var/www/html APP_URL=http://localhost ESLINT_DISABLE=true npm --prefix /var/www/html/vendor/shopware/storefront/Resources/app/storefront run hot

stop-watch-storefront: ## Reverts everything back to normal operation
	@rm -rf /var/www/html/.htaccess.watch

# ---------------------------------------------------------------------------------------------

watch-admin: ## starts watcher for admin at http://localhost:8888
	cd /var/www/html && ./bin/build-administration.sh
	cd /var/www/html && php bin/console bundle:dump
	cd /var/www/html && PROJECT_ROOT=/var/www/html APP_ENV=dev PORT=8888 HOST=0.0.0.0 ENV_FILE=/var/www/html/.env ESLINT_DISABLE=true npm run --prefix /var/www/html/vendor/shopware/administration/Resources/app/administration dev -- http://0.0.0.0


https://github.com/dockware/dockware/blob/master/template/assets/assets/shopware6/makefile.twig

**Comment by martinweinmayr** [comment]
ok i might was wrong, but im getting closer :-)

**Comment by hagmann** [comment]
@martinweinmayr I recreated the docker instance but unfortunately I receive the same results. My makefile looks like this:
![image](https://user-images.githubusercontent.com/11349379/93598718-63663800-f9bd-11ea-9c86-8d36c30403ce.png)
(sorry about the bad contrast)

**Comment by martinweinmayr** [comment]
i have now the same output as you, with the port 9998 and so on.
But storefront works well. and i see changes emediatly.

We will investigate it a bit deeper on monday. To also get fonts working.

**Comment by hagmann** [comment]
@martinweinmayr Awesome, thank you :)

**Comment by martinweinmayr** [comment]
ok found it ...

you have to manually call bin/console theme:compile 
that the default assets getting coppied to the assets folder...

**Comment by LaurensGoedel** [comment]
We have the same issue sinsce last friday after we updated to 6.3.1.0. We tried to call the scripts manually as @hagmann mentioned, but didnt work.   
We tried the bin/console theme:compile too, but alsno won't solve the issue.

For now we added 9998:9998 in our docker-compose file and open the frontend on that port. That seems the only working solution for now for us.

**Comment by martinweinmayr** [comment]
@LaurensGoedel do you also have windows?

today we will release all new versions of shopware for dockware and than normaly everything should work as normal ...

**Comment by LaurensGoedel** [comment]
@martinweinmayr yes, win10 and we use Docker Desktop and WSL to run

That sounds good, looking forward for it!

**Comment by martinweinmayr** [comment]
so finally we fixed it!

solution: in your current setup just update the /var/www/html/.htaccess.watch and add the following line
`RequestHeader add hot-reload-port 9999`

if your fonts a re missing just 
`bin/console theme:compile`

Later this day we will update all dockware images ✌️

**Comment by LaurensGoedel** [comment]
My coworker just encountered the same issue, and your solution for the .htaccess.watch file solves the problem.

**Comment by martinweinmayr** [comment]
as we have now released 6.3.1.0 and 6.3.1.1 the watcher will work again ...
